src/mscorlib/src/System/Number.cs

```diff
@@ -787,10 +777,9 @@ internal unsafe struct NumberBuffer {
 						state |= StateSign | StateParens;
 						number.sign = true;
 					}
-					else if ((currSymbol != null && (next = MatchChars(p, currSymbol)) != null) || (ansicurrSymbol != null && (next = MatchChars(p, ansicurrSymbol)) != null)) {
+					else if ((currSymbol != null && (next = MatchChars(p, currSymbol)) != null)) {
```

Comment:
There is extra pair of round brackets now. Could you please delete it? (Same in the other place.)